Content.Shared/Item/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,263 @@
+
+        _audio.PlayPredicted(component.ActivateSound, uid, user);
+        //Starts the active sound (like humming).
+        component.Stream = _audio.PlayPredicted(component.ActiveSound, uid, user, AudioParams.Default.WithLoop(true))?.Entity;
+
+        return true;
+    }
+
+    public bool TryDeactivate(EntityUid uid, EntityUid? user = null, ItemToggleComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return false;
+
+        if (!component.Activated)
+            return true;
+
+        var attempt = new ItemToggleDeactivateAttemptEvent();
+        RaiseLocalEvent(uid, ref attempt);
+
+        if (attempt.Cancelled)
+            return false;
+
+        Deactivate(uid, component);
+
+        var ev = new ItemToggleDeactivatedEvent();
+        RaiseLocalEvent(uid, ref ev);
+
+        _audio.PlayPredicted(component.DeactivateSound, uid, user);
+        //Stops the active sound (like humming).
+        component.Stream = _audio.Stop(component.Stream);
+
+        return true;
+    }
+
+    //Makes the actual changes to the item.
+    private void Activate(EntityUid uid, ItemToggleComponent component)
+    {
+        component.Activated = true;
+
+        UpdateItemComponent(uid, component);
+        UpdateWeaponComponent(uid, component);
+        UpdateAppearance(uid, component);
+        UpdateLight(uid, component);
+
+        var ev = new ItemToggleActivatedServerChangesEvent();
+        RaiseLocalEvent(uid, ref ev);
+
+        Dirty(uid, component);
+    }
+
+    private void Deactivate(EntityUid uid, ItemToggleComponent component)
+    {
+        component.Activated = false;
+
+        UpdateItemComponent(uid, component);
+        UpdateWeaponComponent(uid, component);
+        UpdateAppearance(uid, component);
+        UpdateLight(uid, component);
+
+        var ev = new ItemToggleDeactivatedServerChangesEvent();
+        RaiseLocalEvent(uid, ref ev);
+
+        Dirty(uid, component);
+    }
+
+
+    private void TurnOffonUnwielded(EntityUid uid, ItemToggleComponent component, ItemUnwieldedEvent args)
+    {
+        if (component.Activated)
+            TryDeactivate(uid, component: component);
+    }
+
+    private void TurnOnonWielded(EntityUid uid, ItemToggleComponent component, ref ItemWieldedEvent args)
+    {
+        if (!component.Activated)
+            TryActivate(uid, component: component);
+    }
+
+
+    /// <summary>
+    /// Used to update item appearance.
+    /// </summary>
+    private void UpdateAppearance(EntityUid uid, ItemToggleComponent component)
+    {
+        if (!TryComp(uid, out AppearanceComponent? appearanceComponent))
+            return;
+
+        _appearance.SetData(uid, ToggleableLightVisuals.Enabled, component.Activated, appearanceComponent);
+        _appearance.SetData(uid, ToggleVisuals.Toggled, component.Activated, appearanceComponent);
+        _appearance.SetData(uid, WelderVisuals.Lit, component.Activated, appearanceComponent);
```

Comment:
change welder visualizer to use generic toggle enum and remove this

Reply:
Done. In addition, since apparently lighters used a lot of welder code, I've ported those over too.
... and added zippo lighters, they were just waiting to be set up, had the sprites already there.